crates/storage/provider/src/providers/database/provider.rs

```diff
@@ -428,8 +428,19 @@ impl<'this, TX: DbTxMut<'this> + DbTx<'this>> DatabaseProvider<'this, TX> {
             .map(|(id, tx)| (id, tx.into()))
             .collect::<Vec<(u64, TransactionSigned)>>();
 
-        let senders =
-            self.get_or_take::<tables::TxSenders, TAKE>(first_transaction..=last_transaction)?;
+        let senders = match self
+            .get_or_take::<tables::TxSenders, TAKE>(first_transaction..=last_transaction)
+        {
+            Ok(senders) => senders,
```

Comment:
depending on the pruning configuration, there might be senders but not for all transactions. should account for that